File: Models/ActionAccess.cs
#nullable disable

namespace CRMService.Models
{
    public class ActionAccess
    {
        public int Id { get; set; }
        public int RoleId { get; set; }
        public int UserId { get; set; }
        public UserRole RootRole { get; set; }
        public User RootUser { get; set; }
        public bool IsDocumentEditor { get; set; }
        public bool IsDocumentReader { get; set; }
        public bool IsDocumentDeleter { get; set; }
        public bool IsDocumentAccessManager { get; set; }
        public int DocumentId { get; set; }
        public Document RootDocument { get; set; }

    }
    enum DocumentAction
    {
        Upload,
        Edit,
        Delete
    }
}

File: ClientApp/src/app/api/models/action-access.ts
/* tslint:disable */
/* eslint-disable */
import { Document } from '../models/document';
import { User } from '../models/user';
import { UserRole } from '../models/user-role';
export interface ActionAccess {
  documentId?: number;
  id?: number;
  isDocumentAccessManager?: boolean;
  isDocumentDeleter?: boolean;
  isDocumentEditor?: boolean;
  isDocumentReader?: boolean;
  roleId?: number;
  rootDocument?: Document;
  rootRole?: UserRole;
  rootUser?: User;
  userId?: number;
}
